ddtrace/internal/core/crashtracking.py

```diff
@@ -1,68 +1,160 @@
+        stacktrace_resolver = StacktraceCollection.Disabled
+    elif crashtracker_config.stacktrace_resolver == "fast":
+        stacktrace_resolver = StacktraceCollection.WithoutSymbols
     elif crashtracker_config.stacktrace_resolver == "safe":
-        crashtracker.set_resolve_frames_safe()
+        stacktrace_resolver = StacktraceCollection.EnabledWithSymbolsInReceiver
+    elif crashtracker_config.stacktrace_resolver == "full":
+        stacktrace_resolver = StacktraceCollection.EnabledWithInprocessSymbols
     else:
-        crashtracker.set_resolve_frames_disable()
+        # This should never happen, as the value is validated in the crashtracker_config
+        # module.
+        print(f"Invalid stacktrace_resolver value: {crashtracker_config.stacktrace_resolver}")
+        stacktrace_resolver = StacktraceCollection.EnabledWithInprocessSymbols
 
-    if crashtracker_config.stdout_filename:
-        crashtracker.set_stdout_filename(crashtracker_config.stdout_filename)
-    if crashtracker_config.stderr_filename:
-        crashtracker.set_stderr_filename(crashtracker_config.stderr_filename)
+    # Create crashtracker configuration
+    config = CrashtrackerConfiguration(
+        [],  # additional_files
+        crashtracker_config.create_alt_stack,
+        crashtracker_config.use_alt_stack,
+        5000,  # timeout_ms
+        stacktrace_resolver,
+        crashtracker_config.debug_url or agent_config.trace_agent_url,
+        None,  # unix_socket_path
+    )
 
-    # Add user tags
-    for key, value in crashtracker_config.tags.items():
-        add_tag(key, value)
+    # Create crashtracker receiver configuration
+    receiver_config = CrashtrackerReceiverConfig(
+        [],  # args
+        # Need to set PATH to find the right python binary
+        os.environ.copy(),  # env
+        crashtracker_exe_py,  # path_to_receiver_binary
+        crashtracker_config.stderr_filename,
+        crashtracker_config.stdout_filename,
+    )
+
+    tags = _get_tags(additional_tags)
+
+    metadata = CrashtrackerMetadata("dd-trace-py", version.get_version(), "python", tags)
+
+    return config, receiver_config, metadata
 
-    if profiling_config.enabled:
-        add_tag("profiler_config", config_str(profiling_config))
 
-    # Only start if it is enabled
-    if crashtracker_config.enabled:
-        return crashtracker.start()
-    return False
+def is_started() -> bool:
+    if not is_available:
+        return False
+    return crashtracker_status() == CrashtrackerStatus.Initialized
+
+
+def start(additional_tags: Optional[Dict[str, str]] = None) -> bool:
+    if not is_available:
+        return False
+    if not crashtracker_config.enabled:
+        return False
+
+    config, receiver_config, metadata = _get_args(additional_tags)
+    if config is None or receiver_config is None or metadata is None:
+        print("Failed to start crashtracker: failed to construct crashtracker configuration")
+        return False
+
+    try:
```

Comment:
move the try around the whole function? might as well try to capture `_get_args` as well?

maybe overkill, but safest?

no strong opinion, random thought, feel free to resolve/ignore this.